diff --git a/tabbed.c b/tabbed.c
index d30206b..70642cb 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -308,6 +308,7 @@ drawbar(void) {
 	unsigned long *col;
 	int c, fc, width, n = 0;
 	char *name = NULL;
+	char tabtitle[256];
 
 	if(nclients == 0) {
 		dc.x = 0;
@@ -353,7 +354,9 @@ drawbar(void) {
 		} else {
 			col = dc.norm;
 		}
-		drawtext(clients[c]->name, col);
+		snprintf(tabtitle, sizeof(tabtitle), "%d: %s",
+		         c + 1, clients[c]->name);
+		drawtext(tabtitle, col);
 		dc.x += dc.w;
 		clients[c]->tabx = dc.x;
 	}
